woocommerce/payment-gateway/payment-tokens/class-sv-wc-payment-gateway-payment-tokens-handler.php

```diff
@@ -363,33 +378,42 @@ public function delete_token( $user_id, SV_WC_Payment_Gateway_Payment_Token $tok
 			$environment_id = $this->get_environment_id();
 		}
 
-		// get existing tokens
-		$tokens = $this->get_tokens( $user_id, array( 'environment_id' => $environment_id ) );
+		// always clear the transient
+		$this->clear_transient( $user_id );
```

Comment:
I think it would be good to clear the transient on `add_token()` and `update_token()` as well. The transient is short lived, but is still possible to get an outdated copy of the tokens in the next request.